apps/rush-lib/src/cli/actions/ChangeAction.ts

```diff
@@ -353,14 +371,20 @@ export class ChangeAction extends BaseRushAction {
     });
   }
 
-  private _hasProjectChanged(changedFolders: (string | undefined)[], projectFolder: string): boolean {
-    for (const folder of changedFolders) {
-      if (folder && Path.isUnderOrEqual(folder, projectFolder)) {
-        return true;
-      }
+  private _hasProjectChanged(
+    changedFiles: string[],
+    projectFolder: string,
+    getChangeIgnorePatterns: (fsPath: string) => string[]
+  ): boolean {
+    const relevantFiles: string[] = changedFiles.filter((file) => Path.isUnderOrEqual(file, projectFolder));
+
+    const ignoreConfig: Ignore = ignore();
+
+    for (const ignorePattern of flatMap(relevantFiles.map(path.dirname), getChangeIgnorePatterns)) {
```

Comment:
It's almost certainly worth deduplicating intermediary values to improve on performance.
```
const foldersToConsider: Set<string> = new Set(relevantFiles.map(path.dirname));
for (const folder of foldersToConsider) {
  const parent: string = path.dirname(folder);
  if (/* is parent in project */) {
    foldersToConsider.add(parent);
  }

  // Can lookup the local ignore file and add the patterns here, including relevant prefixing.
}